src/com/hms/flexyosisoftconnector/Tag.java

```diff
@@ -120,6 +170,264 @@ public void recordTagValue(String time)
       }
    }
 
+   /**
+    * exportBlockDescriptor
+    * send an EBD command with a start and end time and a tag name.
+    * start time is stored in an external file, end time is current time.
+    */
+   public void exportBlockDescriptor()
+   {
+      // time pointers to keep track of start and end of queue
+      String startTime = "null";
+      String endTime = "null";
+
+      // get the time pointer to use in EBD call for the start of historical logs
+      startTime = readTimePointerFile();
+
+      // check that the error value did not get returned. Log if it did, and do not make EBD call.
+      if (startTime.equalsIgnoreCase("-1")) {
+         Logger.LOG_ERR("EBD Error: Start time not set after reading time pointer file.");
+         return;
+      }
+
+      // end time should be the current time to always keep it up to date
+      endTime = convertTimeIntHelper(System.currentTimeMillis());
+
+      String tagName = getTagName();
+
+      try {
+          // https://ewonsupport.biz/ebd/ <- has a nice EBD string builder. :)
+          Exporter exporter = new Exporter("$dtHL$ftT$st"+startTime+"$et"+endTime+"$fnvalues$tn"+tagName);
+          exporter.ExportTo("file:///"+tagDataFileName);
+          exporter.close();
+      }catch (IOException ioe){
+          Logger.LOG_ERR(" error in export block descriptor");
+      }
+   }
+
+   /**
+    * parse a semi colon seperated file for the 'timeInt' and 'value' fields
+    */
+   public void parseTagFile()
+   {
+
+       BufferedReader reader;
+       try {
+           reader = new BufferedReader(new FileReader(tagDataFileName));
+           // read the first line and do nothing with it
+           String line = reader.readLine();
+           if (line != null)

```

Comment:
in this project and all projects we have tried to avoid multiline conditionals without {}